packages/cli/src/platform/_lib/utils.ts

```diff
@@ -63,3 +65,32 @@ export const generateConnectionString = (apiKey: string) => {
   url.searchParams.set('api_key', apiKey)
   return bold(url.href)
 }
+
+export const poll = async <F extends () => Promise<R>, R>(
+  fn: F,
+  until: (res: R) => boolean,
+  waitMs: number,
+  timeoutMs: number,
+  message?: string,
+) => {
+  const endTime = new Date().getMilliseconds() + timeoutMs
+  const waitTime = new Date().getMilliseconds() + waitMs
+
+  const wait = () =>
+    new Promise((resolve) => {
+      setTimeout(resolve, waitMs)
+    })
+
+  let result = await fn()
+  while (!until(result)) {
+    if (waitTime > endTime) {
```

Comment:
`waitTime` needs to change in every loop to be "current timestamp + `waitMs`", otherwise it's always "poll start + `waitMs`"